### PR TITLE
chore(core): add missing Assert.fail() calls to failure scenario tests

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Also changes "Cannot lock table" `CairoException` thrown by REINDEX command to a non-critical one.

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `build`, `ci`, `chore`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8e53e4f7858790f9ba869b861216d198b9108cf2
</p>

File: core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java [hunk 1]
```diff
@@ -746,6 +743,7 @@ public void testAttachPartitionsWithIndexedSymbolsValueMatch() throws Exception
         });
     }
 
+    @Ignore
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java:746]: @marregui could you help me with updating this test to make it successfully passing? I wonder how I should update src and dst tables to get a column mismatch error of any kind.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/RebuildColumnBase.java|1|1|100.00%|

File: core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java (hunk 1 above)
Comment (marregui) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java:746]: yes

Comment (marregui) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java:746]: the test is correct, it passes.

The idea of the test is that we create two tables of different name, but same id. Other than the name the tables are identical, so it means we can use the metadata of one from the other.

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java:746]: The expected exception is not thrown here, so this test fails on my machine. Am I missing something?

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java]: I've reenabled the test in 91d338aeffc7efe59bd712743d329295b498f8d9 to demonstrate the failure.

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java]: BTW you should be looking at `testAttachPartitionsWithSymbolsValueDoesNotMatch`. Based on your test scenario description, I assume that you were looking at another test.

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java]: Nevermind, I've updated the test in 27e53ab2f2f5cce3b01884b0a8babd407605b5ea

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/RebuildColumnBase.java|1|1|100.00%|

**Review by marregui — APPROVED** [review]
LGTM

**Comment by puzpuzpuz** [comment]
@marregui thanks for the review!

**Review by marregui — APPROVED** [review]
LGTM

**Pull request merged**